subsys/net/ip/Kconfig

```diff
@@ -589,10 +589,10 @@ config NET_SLIP_TAP
 	bool "TAP SLIP driver"
 	depends on NET_QEMU_SLIP
 	depends on NET_NATIVE
-	select SLIP
```

Comment:
This breaks the default configuration for practically any networking sample that can run on qemu, it can't go as it is. If we want to give up on `select`, it's needed to update the default value of `SLIP` config, likely set to yes if `NET_QEMU_SLIP` is enabled.

Reply:
Same for the `NET_DRIVERS`, it seems it should also be enabled by default for qemu networking.

Reply:
@rlubos  interesting, seems I missed something.
How come networking samples are not failing with twister when `SLIP` is disabled, even when running the integration tests ?

Have been running complete CI for boards, and integration tests for `qemu_x86`.

A re-run just for `tests/net`.
```
$ ./scripts/twister -c -N --inline-logs -T tests/net/ -p qemu_x86 -G
ZEPHYR_BASE unset, using "/projects/github/ncs/zephyr"
Deleting output directory /projects/github/ncs/zephyr/twister-out
INFO    - Zephyr version: v2.7.0-rc1-140-g76dc66548970
INFO    - JOBS: 16
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testcase list...
INFO    - 152 test scenarios (152 configurations) selected, 59 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:   93/  93  100%  skipped:   63, failed:    0
INFO    - 89 of 89 test configurations passed (100.00%), 0 failed, 63 skipped with 0 warnings in 94.45 seconds
INFO    - In total 725 test cases were executed, 1653 skipped on 1 out of total 394 platforms (0.25%)
INFO    - 81 test configurations executed on platforms, 8 test configurations were only built.
INFO    - Saving reports...
```

I would expect tests that require `CONFIG_SLIP=y` to fail in such case.
But all `tests/net` passes both on main and with this PR, even test cases where `CONFIG_SLIP=y` in main but not in PR.

Reply:
As far as I understand, the networking tests executed by twister are self-contained, i.e. using the loopback interface, so they don't need the actual driver in order to execute. But when it comes to the samples, where the actual network communication is expected, they won't work w/o SLIP (by default). 

The tricky part is, that even w/o SLIP being enabled the samples build just fine, there's no compilation error. But when you run the sample, the system complains that there is no network interface to work with.

Reply:
> But when it comes to the samples, where the actual network communication is expected, they won't work w/o SLIP (by default).

yes, but this line in the setting that was changed from `select` to `depends` looks like this:
```
config NET_SLIP_TAP
	bool "TAP SLIP driver"
	depends on NET_QEMU_SLIP
	depends on NET_NATIVE
	depends on SLIP
```
so it has a dependency to `NET_QEMU_SLIP`, which is defined here: 
https://github.com/zephyrproject-rtos/zephyr/blob/00f5f3fab4c48097b1c2bc49698137fc50bcc90d/subsys/net/ip/Kconfig#L74

and is under a choice which `depends on QEMU_TARGET`:
https://github.com/zephyrproject-rtos/zephyr/blob/00f5f3fab4c48097b1c2bc49698137fc50bcc90d/subsys/net/ip/Kconfig#L64-L74

so my change is only applicable for Qemu.

And twister launches Qemu in order to run those tests:
```
2021-09-22 17:03:37,246 - twister - DEBUG - Running qemu_x86/tests/net/neighbor/net.neighbor.preempt (qemu)
2021-09-22 17:03:37,247 - twister - DEBUG - QEMU (637692): Booting from ROM..*** Booting Zephyr OS build v2.7.0-rc1-140-g76dc66548970  ***
2021-09-22 17:03:37,248 - twister - DEBUG - QEMU (637692): Running test suite test_arp_fn
2021-09-22 17:03:37,248 - twister - DEBUG - QEMU (637692): ===================================================================
2021-09-22 17:03:37,248 - twister - DEBUG - QEMU (637692): START - test_arp
2021-09-22 17:03:37,248 - twister - DEBUG - Spawning QEMUHandler Thread for qemu_x86/tests/net/neighbor/net.neighbor.preempt
2021-09-22 17:03:37,249 - twister - DEBUG - Running qemu_x86/tests/net/ieee802154/fragment/net.ieee802154.fragment (qemu)
2021-09-22 17:03:37,250 - twister - DEBUG - Spawning QEMUHandler Thread for qemu_x86/tests/net/ieee802154/fragment/net.ieee802154.fragment
2021-09-22 17:03:37,251 - twister - DEBUG - Running qemu_x86/tests/net/igmp/net.igmp.preempt (qemu)
2021-09-22 17:03:37,254 - twister - DEBUG - QEMU (637692): PASS - test_arp in 0.7 seconds
2021-09-22 17:03:37,255 - twister - DEBUG - QEMU (637692): ===================================================================
2021-09-22 17:03:37,255 - twister - DEBUG - QEMU (637692): Test suite test_arp_fn succeeded
2021-09-22 17:03:37,255 - twister - DEBUG - QEMU (637692): ===================================================================
2021-09-22 17:03:37,255 - twister - DEBUG - QEMU (637692): PROJECT EXECUTION SUCCESSFUL
....
```

so if qemu anyway just uses the loopback device, then why is the `NET_QEMU_SLIP` flag there at all ?

>  But when it comes to the samples, where the actual network communication is expected, they won't work w/o SLIP (by default).

Are you having some other samples in mind besides `tests/net` ?
Notice, the `select SLIP` we are discussing here is only applicable to the `QEMU_TARGET` due to dependency chain.
This change has no impact on any real board.

> The tricky part is, that even w/o SLIP being enabled the samples build just fine, there's no compilation error. But when you run the sample, the system complains that there is no network interface to work with.

yes, which is exactly why i'm running the samples using qemu.

Reply:
> Are you having some other samples in mind besides tests/net ?

Every sample under `samples/net`.

> so if qemu anyway just uses the loopback device, then why is the NET_QEMU_SLIP flag there at all ?

It uses a loopback interface for tests. But if you run some actual sample, that needs to communicate with some other device in the network, you need some emulated driver (and by default SLIP was used).

> Notice, the select SLIP we are discussing here is only applicable to the QEMU_TARGET due to dependency chain.
This change has no impact on any real board.

And this is perfectly fine, real boards should use some actual networking hardware they are providing. But having a possibility to run networking samples in emulated environment like qemu easily (i.e. with the default `prj.conf`, w/o any extra hassle) is a must-have feature IMO, dropping such support is a serious regression.

Reply:
@rlubos thanks for the detailed explanation.

I have pushed an update that changes some Kconfig slightly, but in a way so that things should work as desired now.

Reply:
Thanks, it seems to work fine now